contrib/go/src/python/pants/contrib/go/tasks/go_buildgen.py

```diff
@@ -215,6 +221,15 @@ def register_options(cls, register):
             help="An optional extension for all materialized BUILD files (should include the .)",
         )
 
+        register(
+            "--remote-root",
+            default=None,
```

Comment:
It seems like it would be good to establish (or keep) a precedent instead of forcing every go user with external deps to add a pants.toml value for this. In fact, this is a breaking change as is with no deprecation cycle so I think adding the likely most common default - probably 3rdparty/go, at least with a dep cycle before removing, is required. This still breaks anyone out there who relied on implicit `3rd_party/go`, `thirdparty/go` or `third_party/go` though.

Reply:
I've added an attempt to infer which of the default options it is. It'll still break if someone set a custom remote root in `[source]`, but I'm skeptical anyone has. And at least there's a sensible error message.